### PR TITLE
Fix #661: Removed all UIWebView code.

### LINKED ISSUE #661: Remove all UIWebView code?
People has started receiving this warning when submitting apps that have references to UIWebView. 

> ITMS-90809: Deprecated API Usage - Apple will stop accepting submissions of apps that use UIWebView APIs.

Doesn't matter if they are using just WKWebView, as long as Apple sees "UIWebView" in the app code, they show the warning, and cordova-ios is full of "UIWebView" references. (Even react native, flutter and Capacitor users have received the warning)

**There is currently no solution to this problem for Cordova apps.**

So, instead of integrating WKWebView and making it possible to switch from UIWebView to WKWebView on runtime as it was planned, maybe we should just remove all UIWebView code on next major release?

~~There is no deadline, I've contacted an Apple developer that has been tweeting about moving to WKWebView to see if he can provide more information, will update if I get a response.~~

Apple finally announced the [deadline](https://developer.apple.com/news/?id=12232019b):

> The App Store will no longer accept new apps using UIWebView as of April 2020 and app updates using UIWebView as of December 2020.

### PR DESCRIPTION
**Pull request by bpresles**
[Build Status](https://travis-ci.org/bpresles/cordova-ios)

Removed all UIWebView related code.
Replaced default engine by CDVWKWebViewEngine.

### Motivation and Context

I've tested the pull request from https://github.com/apache/cordova-ios/pull/662, but the AppStore Connect still reported "ITMS-90809: Deprecated API Usage - Apple will stop accepting submissions of apps that use UIWebView APIs." with it (I tripled check that the WKWebViewOnly preference was correctly taken in account).

Also I don't think that doing a half way solution at compile time, instead of removing the UIWebView code altogether, is a good solution.
If the UIWebView is deprecated, than remove the code altogether as it'll be useless in the near future. People wanting to still use UIWebView can either use a UIWebView engine as a plugin or use older version of cordova-ios.

https://github.com/apache/cordova-ios/issues/661

### Description
Removed all UIWebView code and replaced default engine by CDVWKWebViewEngine from cordova-plugin-wkwebview-engine (making it useless now).

### Checklist

- [x] I've run the tests to see all new and existing tests pass
- [x] I added automated test coverage as appropriate for this change
- [ ] Commit is prefixed with `(platform)` if this change only applies to one platform (e.g. `(android)`)
- [x] If this Pull Request resolves an issue, I linked to the issue in the text above (and used the correct [keyword to close issues using keywords](https://help.github.com/articles/closing-issues-using-keywords/))
- [ ] I've updated the documentation if necessary

### CODE REVIEW

**Comment by fjaeger** [comment]
I don't see why #662 should not work if the preference is set correctly. All `UIWebview`-related code is encapsulated in preprocessor conditions. 
Maybe you have some other plugin in your Cordova project that still contains traces to `UIWebView`?

**Comment by bpresles** [comment]
> I don't see why #662 should not work if the preference is set correctly. All `UIWebview`-related code is encapsulated in preprocessor conditions.
> Maybe you have some other plugin in your Cordova project that still contains traces to `UIWebView`?

I added this line on the iOS platform section in my config.xml:

`<preference name=“WKWebViewOnly” value=“true” />`

Then I removed the iOS platform all together and added it back, and made sure now XCode that the `WK_WEB_VIEW_ONLY` setting was added  in the user defined build settings and set to `1` which was the case.

Then I cleaned the build folder and build and run the app which was working just fine using `WKWebview` from `cordova-plugin-wkwebview-engine`.

So I make an Archive and submitted the app to AppStore Connect. And I still got the warning. I tried to completely remove the `node_modules` folder as well as the `plugins` folder to make sure that they geet fetched again, and remove iOS platform than readded, but it was no better (the app worked fine but AppStore Connect still reported the UIWebView deprecation warning).

Then I worked on removing all reference of `UIWebView` in `cordova-ios` and removed the embedded CDVUIWebViewEngine, at first still using the `cordova-plugin-wkwebview-engine` plug-in without embedding it, and it was only then I could send an archive that didn’t trigger the deprecation warning (so without changing anything in my installed plugins). And then I worked on embedding the `cordova-ios` and removed the embedded CDVUIWebViewEngine, at first still using the `cordova-plugin-wkwebview-engine` directly in `cordova-ios` as default engine.

Also I don’t think that making the wkwebview only just an optional compile time preference isn’t a good idea. Because as Apple will not just deprecate the UIWebView APIs but reject all apps using them in the future, it’s not very useful to keep this old UIWebView based engine as default.
If anyone want to still use UIWebView they can simply use previous versions of `cordova-ios` that are using it as default engine.

**Comment by EdwinRozario** [comment]
Will this PR be merged anytime soon or are we all going to install @bpresles 's fork?

**Comment by timbru31** [comment]
We haven't decided yet how to handle Apple's deprecation warning, please see (and participate :)) in the discussion here: https://github.com/apache/cordova-discuss/pull/110/

**Comment by rajashekaranugu** [comment]
https://blog.ionicframework.com/understanding-itms-90809-uiwebview-api-deprecation/
 Ionic blog

**Comment by davidecastello** [comment]
Any news on this? This would be really appreciated, thanks.

**Comment by adminy** [comment]
This seems to be mostly done, why hasn't this been merged yet? Apple has been giving the notice for a while now.

File: CordovaLib/Classes/Private/Plugins/CDVIOSWKWebViewEngine/CDVIOSWKProcessPoolFactory.h [hunk 1]
```diff
@@ -6,9 +6,9 @@
  to you under the Apache License, Version 2.0 (the
  "License"); you may not use this file except in compliance
  with the License.  You may obtain a copy of the License at
-
+
```
Comment (brodybits) [inline CordovaLib/Classes/Private/Plugins/CDVIOSWKWebViewEngine/CDVIOSWKProcessPoolFactory.h]: Please remove the extra space added here and on line 11

**Comment by brodybits** [comment]
Thanks guys for the contribution and the followup. This is under discussion in apache/cordova-discuss#110, and there are multiple blocking factors right now:

- we still need some more time to reach agreement, and 100% consensus would be ideal
- we need to have a committer review the changes
- committers including myself are pretty overloaded

While I would agree that this is pretty urgent, we would appreciate a little bit of slack and some benefit of the doubt. I would rephrase the followup question something like this: "What is the status? This looks pretty urgent."

Guys please DO feel free to follow up with us from time to time. I did just push the discussion in this comment: https://github.com/apache/cordova-discuss/pull/110#issuecomment-548877697

I would also recommend you guys consider following up via mailing list or slack, our contact is here: https://cordova.apache.org/contact/

**Comment by erisu** [comment]
@bpresles Can you rebase this PR with master and resolve the conflicts?

Also, is it possible to renamed the classes with the prefix of `CDVWebView` instead of `CDVIOSWK`? 

Since `UIWebView` is deprecated and wont be accepted in the AppStore for new apps starting in April, it means there will eventually be only one WebView (WKWebView).

@dpogue @jcesarmobile Do you have any opinions on using `CDVWebView` as the prefix instead?

**Comment by jcesarmobile** [comment]
I think at this point is simpler to create a new PR, since the conflicts are due to the conditional compiler flag and it's going to take more time to sync all those files than creating a new PR.
Or we can revert the conditional flag PR since it's not going to be part of next release

**Comment by TakaKeiji** [comment]
Any news about this?
I'm trying to upload my app to TestFlight and since mid april 2020 the UIWebView deprecation restriction is active and rejecting any app with references to UIWebView

**Comment by breautek** [comment]
@TakaKeiji 

I cannot comment about when 6.0 will be released but you can handle the uiwebview issue on 5.1.1

see https://cordova.apache.org/howto/2020/03/18/wkwebviewonly.html to learn how.

**Comment by TakaKeiji** [comment]
> @TakaKeiji
> 
> I cannot comment about when 6.0 will be released but you can handle the uiwebview issue on 5.1.1
> 
> see https://cordova.apache.org/howto/2020/03/18/wkwebviewonly.html to learn how.

Thanks for the quick response, i'll try your recommendation later!

**Comment by TakaKeiji** [comment]
> @TakaKeiji
> 
> I cannot comment about when 6.0 will be released but you can handle the uiwebview issue on 5.1.1
> 
> see https://cordova.apache.org/howto/2020/03/18/wkwebviewonly.html to learn how.

Working correctly now, the build was accepted correctly, thanks for the info, anyway i'm still having issues with the oficial archived PayPal mobile sdk cordova plugin, so i'm looking a solution for this now...

**Comment by i-need-to-tell-you-something** [comment]
> > @TakaKeiji
> > I cannot comment about when 6.0 will be released but you can handle the uiwebview issue on 5.1.1
> > see https://cordova.apache.org/howto/2020/03/18/wkwebviewonly.html to learn how.
> 
> Working correctly now, the build was accepted correctly, thanks for the info, anyway i'm still having issues with the oficial archived PayPal mobile sdk cordova plugin, so i'm looking a solution for this now...

I tried to find the source, but couldn't - I _just_ read on of these issues that Paypal SDK is deprecated and users should move to BrainTree or one other solution.

**Comment by breautek** [comment]
The paypal plugin provided by PayPal itself appears to be referencing `UIWebView` as confirmed by the user at https://github.com/apache/cordova-ios/issues/854#issuecomment-629819404

As the PayPal deprecated that SDK, and the sdk appears to be closed-source, the only solution is to move away from using the paypal sdk. Their deprecation notice can be found on their [README](https://github.com/paypal/PayPal-Cordova-Plugin/blob/master/README.md), which contains this text:

> Please use [Braintree Direct](https://www.braintreepayments.com/ca/products/braintree-direct?referrer=https%3A%2F%2Fgithub.com%2Fpaypal%2FPayPal-Cordova-Plugin%2Fblob%2Fmaster%2FREADME.md) in [supported countries](https://www.braintreepayments.com/ca/country-selection?referrer=https%3A%2F%2Fgithub.com%2Fpaypal%2FPayPal-Cordova-Plugin%2Fblob%2Fmaster%2FREADME.md). In other countries, use [Express Checkout](https://developer.paypal.com/docs/accept-payments/express-checkout/ec-braintree-sdk/get-started/) and choose the Braintree SDK integration option.

So specifically, the solution will depend on your situation.

Hope this puts you on the right track, but I'll be locking this because this is not the place for support. If you have further questions on anything cordova related, please use our [slack](http://slack.cordova.io/). For questions regarding the new way to use PayPal, you'll likely be better off going through their support channels.
